### PR TITLE
 [Flex 624] signpost component for clarity

### PR DESCRIPTION
**Pull request by psivakumaran**
Signpost component for clarity-react.

### CODE REVIEW

**Comment by psivakumaran** [comment]
[Signpost_positions.docx](https://github.com/EMCECS/clarity-react/files/3400019/Signpost_positions.docx)

Screenshots for the signpost positions

File: src/forms/signpost/SignPost.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as React from "react";
+import ReactDOM from "react-dom";
```
Comment (ben-schumacher) [inline src/forms/signpost/SignPost.tsx]: Unused import.

File: src/forms/signpost/AxisPosition.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import {SignPostDirection} from "./SignPost";
```
Comment (alexander-rassanov) [inline src/forms/signpost/AxisPosition.tsx]: seems the licence was missed

File: src/forms/signpost/SignPost.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+                            bottom: "auto",
+                            left: "0px",
+                            right: "auto",
+                            transform: transformVal,
+                        }}
+                    >
+                        <div className="signpost-flex-wrap">
+                            <div className="popover-pointer" />
+                            <div className="signpost-content-header">
+                                <Button className="signpost-action close" onClick={this.handleButtonClick.bind(this)}>
```
Comment (alexander-rassanov) [inline src/forms/signpost/SignPost.tsx]: Better to create an arrow function that will be bound to "this" of the component. Here you will bind this function to this at each render call

File: src/forms/signpost/AxisPosition.tsx (hunk 2 above)
Comment (psivakumaran) [reply src/forms/signpost/AxisPosition.tsx]: Added the license, thanks for noticing

Comment (psivakumaran) [reply src/forms/signpost/AxisPosition.tsx]: Added.

File: src/forms/signpost/SignPost.tsx (hunk 3 above)
Comment (psivakumaran) [reply src/forms/signpost/SignPost.tsx]: Done

File: src/forms/signpost/SignPost.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+                            bottom: "auto",
+                            left: "0px",
+                            right: "auto",
+                            transform: transformVal,
+                        }}
+                    >
+                        <div className="signpost-flex-wrap">
+                            <div className="popover-pointer" />
+                            <div className="signpost-content-header">
+                                <Button className="signpost-action close" onClick={() => this.handleButtonClick()}>
```
Comment (alexander-rassanov) [inline src/forms/signpost/SignPost.tsx]: Sorry, probably I was not correct in the explanation. I meant that if you have a pattern for defining function via arrow function you would use onClick={this.handleButtonClick}. Where handleButtonClick = () => {......}

You can read some ideas here https://github.com/facebook/react/issues/9851

But if you choose use bind. the previous code is correct, but better the bind call move to constructor() like in the link above

Comment (psivakumaran) [reply src/forms/signpost/SignPost.tsx]: Sorry I misunderstood, corrected the function.
Thank you.

File: src/forms/signpost/SignPost.tsx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        const {direction} = this.props;
+        let transformVal =
+            direction && calculateAxisPosition(parentHeight, parentWidth, childHeight, childWidth, direction);
+        this.setState({transformVal: transformVal});
+    }
+
+    render() {
+        const {isOpen, transformVal} = this.state;
+        const {direction, style, children, shape, size} = this.props;
+        let iconShape = shape ? shape : "dell-alert-info";
```
[the remark this answers is not in the document]
Comment (psivakumaran) [reply src/forms/signpost/SignPost.tsx]: Done.

**Pull request merged**